### ISSUE TITLE
Type Error in iOS 10.3 headers should not be undefined

### ISSUE THREAD

**Issue by tengattack** [agraboso/redux-api-middleware#122]
Hi, Apple release iOS 10.3 officially today.
It has some change in Safari.

`fetch` API will throw an Error: `TypeError: undefined is not an object`.
By simply doing this:
```js
fetch('http://google.com', { method: 'GET', credentials: 'omit', headers: undefined })
```

It cause our library can't do api request normally.

I think the one thing we only need to do is removing `headers` property in options when it is `undefined`.

Thanks

**Comment by mikelaurence** [comment]
We just discovered the same problem with Mac Safari 10.1.

Solved with this change in `middleware.js` on line 103 (note the `headers || {}`):

```
// Make the API call
var res = await fetch(endpoint, { method, body, credentials, headers: headers || {} });
```

**Comment by mikelaurence** [comment]
PR submitted.

**Comment by amangeot** [comment]
Hello,
thanks a lot for fixing this issue, you saved us a lot of time too.

I'm trying to use your fix. I installed your branch with
`npm install git://github.com/mikelaurence/redux-api-middleware.git#fix_undefined_headers`
but webpack cannot resolve the module `redux-api-middleware`.

Am I missing something ?

I have done the same for some other pull requests before they were merged, and it is working for these.

**Comment by mikelaurence** [comment]
@amangeot I had trouble installing it as a lib via webpack as well, so I just dumped all the files in my project and referenced them via normal paths. Not as elegant as I'd like, but it works.

**Comment by dimaturishchev** [comment]
Hi @amangeot, the thing is this repository has`prepublish` npm script which builds everything into `lib` folder. You can temporary use my repository as a hack until the fix is merged https://github.com/dimaturishchev/redux-api-middleware

**Comment by rvanmil** [comment]
In case anyone else runs into this and since this module seems abandoned for now and this is not getting merged, a quick workaround for the issue is to add `headers: {}` to your `[CALL_API]` actions which don't specify any headers yet.

**Comment by tanfrank** [comment]
Another option for a quick fix is adding a custom middleware before the apiMiddleware that checks if headers is undefined, and sets it to {}.
 
`const apiAuthMiddleware = store => next => action => {
    const callApi = action[CALL_API];

    if (callApi && !callApi.headers) {
        callApi.headers = {};
    }

    return next(action);
  };`

**Comment by nason** [comment]
A fix was merged in #123 and will be in the next release shortly

**Comment by Bigdragon13th** [comment]
I still got `undefined is not an object` error in version 1.0.3. Is the fix was published yet?
